controller/src/main/java/io/pravega/controller/server/rpc/grpc/impl/GRPCServerConfigImpl.java

```diff
@@ -73,6 +77,11 @@ public GRPCServerConfigImpl(final int port, final String publishedRPCHost, final
         this.authorizationEnabled = authorizationEnabled;
         this.userPasswordFile = userPasswordFile;
         this.tlsEnabled = tlsEnabled;
+        if (tlsProtocolVersion == null) {
+            this.tlsProtocolVersion = Config.TLS_PROTOCOL_VERSION.toArray(new String[Config.TLS_PROTOCOL_VERSION.size()]);
```

Comment:
Like I mentioned earlier, it may be better to just store an immutable array of String and use that instead of creating copies of this array everytime.

Reply:
Why don't you use your `TlsProtocolVersion` class here?

Reply:
Reason not to use TlsProtocolVersion here is to avoid calling `parse` method multiple times. `Tlsprotocolversion` is used only  in Serviceconfig and config classes and from these classes parsed value is passed to `Segmentstore and Controller's(GRPC,REST)` components.